src/prototype_structure.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+/**
+ * Provides structure memory.
+ */
+Object.defineProperty(Structure.prototype, "memory", {
+  get: function () {
+    if(Memory.structures === undefined) {
+      Memory.structures = {};
+    }
+    if(Memory.structures[this.id] === undefined) {
+      Memory.structures[this.id] = {};
+    }
+    return Memory.structures[this.id];
+  },
+  set: function(v) {
+    return _.set(Memory, 'structures.' + this.id, v);
+  },
+  configurable: true,
+  enumerable: false
+});
+
+Structure.prototype.getFutureEnergy = function() {
+  if (this.memory.futureEnergy === undefined) {
+    return 0;
+  }
+  return this.memory.futureEnergy;
+};
+
+Structure.prototype.changeFutureEnergy = function(amount) {
```

Comment:
I think you need to add some kind of decay if for some reason the creep will no deliver, e.g. it dies.

I started to implement a creep.died method (https://github.com/TooAngel/screeps/blob/ce4ee50d81e07f28552b3bf938653f1c2061abff/src/role_nextroomerattack.js#L24) It is not properly tested

But I think I would prefer some kind of decay to handle it on a more general level.

Keep a list of added futures with Game.time when it was added and check / remove on `getFutureEnergy` or something like that.